test-suite/exonum-py-tests/exonum_tests/freeze.py

```diff
@@ -0,0 +1,143 @@
+        cryptocurrency_advanced_config_dict = generate_config(self.network, instances=instances, deploy=False)
+
+        cryptocurrency_advanced_config = Configuration(cryptocurrency_advanced_config_dict)
+        with Launcher(cryptocurrency_advanced_config) as launcher:
+            launcher.deploy_all()
+            launcher.wait_for_deploy()
+            launcher.start_all()
+            launcher.wait_for_start()
+
+        # Check that the service status has been changed for `frozen`.
+        for service in client.public_api.available_services().json()["services"]:
+            if service["spec"]["name"] == "crypto":
+                self.assertEqual(service["status"]["type"], "frozen")
+
+        # Try to create a new wallet. The operation should be failed.
+        with ExonumCryptoAdvancedClient(client) as crypto_client:
+            bob_keys = KeyPair.generate()
+            response = crypto_client.create_wallet(bob_keys, "Bob")
+            self.assertEqual(response.status_code, 400)
+            self.assertEqual(
+                response.json()["title"], "Failed to add transaction to memory pool"  # Cause the service is frozen
+            )
+
+        # Check that we can use service endpoints for data retrieving. Check wallet once again.
+        with ExonumCryptoAdvancedClient(client) as crypto_client:
+            alice_balance = crypto_client.get_balance(alice_keys)
+            self.assertEqual(alice_balance, 100)
+
+    def test_resume_after_freeze_service(self):
+
+        host, public_port, private_port = self.network.api_address(0)
+        client = ExonumClient(host, public_port, private_port)
+
+        # Create wallet
+        with ExonumCryptoAdvancedClient(client) as crypto_client:
+            alice_keys = KeyPair.generate()
+            crypto_client.create_wallet(alice_keys, "Alice")
+            with client.create_subscriber("transactions") as subscriber:
+                subscriber.wait_for_new_event()
+                alice_balance = crypto_client.get_balance(alice_keys)
+                self.assertEqual(alice_balance, 100)
+
+        # Freeze the service
+        instances = {"crypto": {"artifact": "cryptocurrency", "action": "freeze"}}
+        cryptocurrency_advanced_config_dict = generate_config(self.network, instances=instances, deploy=False)
+
+        cryptocurrency_advanced_config = Configuration(cryptocurrency_advanced_config_dict)
+        with Launcher(cryptocurrency_advanced_config) as launcher:
+            launcher.deploy_all()
+            launcher.wait_for_deploy()
+            launcher.start_all()
+            launcher.wait_for_start()
+
+        # Check that the service status has been changed for `frozen`.
+        for service in client.public_api.available_services().json()["services"]:
+            if service["spec"]["name"] == "crypto":
+                self.assertEqual(service["status"]["type"], "frozen")
+
+        # Try to create a new wallet. The operation should be failed.
```

Comment:
Nit: Does it make sense to repeat the steps from the previous test given that freezing the service isn't the focus of *this* test?